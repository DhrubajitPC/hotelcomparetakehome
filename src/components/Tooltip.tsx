import React, { FC, useState } from "react";

type Props = {
  body: HTMLElement;
};

const Tooltip: FC<Props> = ({ body, children }) => {
  const [showTooltip, setShowTooltip] = useState(false);

  return (
    <div
      onMouseEnter={() => setShowTooltip(true)}
      onMouseLeave={() => setShowTooltip(false)}
      style={{
        position: "relative",
      }}
    >
      {children}
      {showTooltip && (
        <div
          style={{
            position: "absolute",
            top: 0,
            left: 0,
            background: "#dfe6e9",
            borderRadius: "5px",
          }}
        >
          {body}
        </div>
      )}
    </div>
  );
};

export default Tooltip;
