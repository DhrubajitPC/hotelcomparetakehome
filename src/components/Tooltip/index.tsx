import React, { FC, ReactElement, useState } from "react";

type Props = {
  body: ReactElement;
};

const Tooltip: FC<Props> = ({ body, children }) => {
  const [showTooltip, setShowTooltip] = useState(false);

  return (
    <div
      data-testid="tooltip-container"
      onMouseEnter={() => setShowTooltip(true)}
      onMouseLeave={() => setShowTooltip(false)}
      style={{
        position: "relative",
      }}
    >
      {children}
      {showTooltip && (
        <div
          style={{
            position: "absolute",
            top: "20px",
            left: 0,
            background: "#dfe6e9",
            borderRadius: "5px",
          }}
        >
          {body}
        </div>
      )}
    </div>
  );
};

export default Tooltip;
