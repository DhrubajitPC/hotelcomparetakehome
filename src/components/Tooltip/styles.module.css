:root {
	--black: #535c68
}

.body {
  position: absolute;
  top: 30px;
  left: -30px;
	padding: 20px;
	border: 1px solid var(--black);
	width: 100px;
	box-shadow: 0 4px 2px -2px var(--black);
	border-radius: 5px;
}
