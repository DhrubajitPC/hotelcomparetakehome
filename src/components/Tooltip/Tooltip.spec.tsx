import { fireEvent, render, screen } from "@testing-library/react";
import Tooltip from ".";
import React, { FC } from "react";

describe("#Tooltip", () => {
  let BodyComponent: FC;
  beforeAll(() => {
    BodyComponent = () => <p>Mock para</p>;
  });

  it("should render correctly", () => {
    const { container } = render(<Tooltip body={<BodyComponent />}></Tooltip>);
    expect(container).toMatchSnapshot();
  });

  it("should show body component when hovered over", () => {
    render(<Tooltip body={<BodyComponent />}></Tooltip>);
    const element = screen.getByTestId("tooltip-container");
    expect(screen.queryByText(/Mock para/i)).toBeNull();
    fireEvent.mouseEnter(element);
    expect(screen.getByText(/Mock para/i)).toBeInTheDocument();
    fireEvent.mouseLeave(element);
    expect(screen.getByText(/Mock para/i)).not.toBeInTheDocument();
  });
});
