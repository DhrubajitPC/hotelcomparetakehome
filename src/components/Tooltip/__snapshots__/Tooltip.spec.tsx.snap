// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelInfo should render correctly 1`] = `
<div>
  <div
    data-testid="tooltip-container"
    style="position: relative;"
  />
</div>
`;
